### PR TITLE
Atmos construction rework again

### LINKED ISSUE #25585: Wrenching left facing gas filter / mixer over an existing pipe switches it to right facing
Pretty self explanatory.

http://c2n.me/3IX233H.png

Left facing Filter / Mixer + wrenched over existing piping =

http://c2n.me/3IX248O.png

### PR DESCRIPTION
**Pull request by duncathan**
major cleanup of `pipe/construction.dm`. behavior is no longer defined by switch statements and if checks on type vars; it is instead handled using OOP principles per code standards

the RPD still needs a major cleanup but that's for a future PR at this point. tgui demoralized me for now.

Fixes #25585

### CODE REVIEW

File: code/modules/atmospherics/machinery/pipes/layermanifold.dm [hunk 1]
```diff
@@ -31,17 +33,7 @@
...
-		all_connected[I] = I
-	for(var/obj/machinery/atmospherics/I in back_nodes)
-		all_connected[I] = I
-	for(var/obj/machinery/atmospherics/I in nodes)
-		all_connected[I] = I
-	var/list/obj/machinery/atmospherics/returnlist = list()
-	for(var/obj/machinery/atmospherics/A in all_connected)
-		returnlist += all_connected[A]
-	return returnlist
+	return front_nodes + back_nodes + nodes
```
Comment (duncathan) [inline code/modules/atmospherics/machinery/pipes/layermanifold.dm]: forgot to mention that i cleaned up this proc while I was here

**Comment by MoreRobustThanYou** [comment]
ffs

**Comment by Cyberboss** [comment]
@duncathan If the hard part of tgui is compiling it check out https://dextraspace.net/TGUICompiler

**Comment by duncathan** [comment]
nah, it's that tgui apparently doesn't have support for `/icon`s and that meant it was useless for the RPD

**Comment by Cyberboss** [comment]
@MrStonedOne made a proc for sending them as base64 I forget how to use it
though

**Comment by duncathan** [comment]
unless tgui already has the ability to decode base64 into icons then I am not touching it

**Comment by Cyberboss** [comment]
Yeah I think some clockwork stuff uses it

**Comment by MrStonedOne** [comment]
it does @duncathan

**Comment by duncathan** [comment]
I ASKED AND NOBODY COULD TELL ME GRRRRR

**Comment by MrStonedOne** [comment]
Thats because its not part of tgui. its part of browsers.

**Comment by MrStonedOne** [comment]
but that's besides the point. the PROPER way to handle icons is to add them to an asset cache then send the asset cache to the player before opening the interface.

check pda code.

Then you can reference the icons as images using 

There is even an easy subtype of asset caches that will preload all icon states in an icon.

I should point out that asset cache was created by /vg/ purely to handle the RPD

**Pull request merged**